File: MedicApp/Models/Account.cs
using MedicApp.Enums;
using MedicApp.Utils;

namespace MedicApp.Models
{
    public class Account
    {
        public Guid Id { get; set; }
        public string Username { get; set; }
        public string Password { get; set; }
        public string Email { get; set; }
        public bool IsActive { get; set; }
        public Roles Role { get; set; }   
        public DateTime CreatedAt { get; set; }
        public bool IsDeleted { get; set; }

        public Account()
        {
            Id = Guid.NewGuid();
            CreatedAt = DateTime.Now;
        }

    }

}

File: Client/src/app/model/Account.ts
import { LoyaltyLevel } from "./LoyaltyLevel";
import { Roles } from "./Roles";


export interface AccountInterface {
  id: number;
  username: string;
  password: string;
  email: string;
  isActive: boolean;
  createdAt: Date;
  isDeleted: boolean;
  loyaltyLevel: LoyaltyLevel;
  role: Roles
}

export class Account implements AccountInterface {
  id: number;
  username: string;
  password: string;
  email: string;
  isActive: boolean;
  createdAt: Date;
  isDeleted: boolean;
  loyaltyLevel: LoyaltyLevel;
  role: Roles

  constructor(obj: AccountInterface) {
    this.id = obj.id;
    this.username = obj.username;
    this.password = obj.password;
    this.email = obj.email;
    this.isActive = obj.isActive;
    this.createdAt = obj.createdAt;
    this.isDeleted = obj.isDeleted;
    this.loyaltyLevel = obj.loyaltyLevel;
    this.role = obj.role;
  }
}
